course_catalog/etl/loaders.py

```diff
@@ -0,0 +1,45 @@
+"""Course catalog data loaders"""
+from django.contrib.contenttypes.models import ContentType
+
+from course_catalog.models import Course, Program, ProgramItem
+
+
+def load_course(course_data):
+    """Load the course into the database"""
+    course_id = course_data.pop("course_id")
+    course, _ = Course.objects.update_or_create(
```

Comment:
The course data should be coming from mitx, and in the case of MM it _should_ exist there before it does in MM (@pdpinch can you confirm?). My plan is to eventually port all the ingestion for mitx, bootcamps, and ocw over to this more straightforward implementation. In those cases `course_data` will have other fields populated for those integrations, and MM becomes a special case where the integration exists only to join the courses under the program, but doesn't do anything additional.

To give a little more context on the direction this is headed, here's how the integration with xpro is going so far, building on the changes introduced in this PR: https://github.com/mitodl/open-discussions/commit/8f925d1532878348a4cc8ba9392a3545a274fb2d

Reply:
I'm not 100% sure what you're asking me here. It's possible for a course to exist on MM before it does on edx.org, but we should consider edx.org to be canonical. In the case that there is a course in MM that does not exist on edx.org it should probably be ignored. 

I can outline a real-life scenario if that level of detail is helpful.

Reply:
Yeah, can you give a concrete example around this? The overlapping integrations here complicate things, so having as much detail on data scenarios is helpful.

Reply:
Actually, if a course exists in MM but not in edx, we can simply leave the `edx_key` blank on the `Course` in MM and that will cause that course to get skipped here. If that's good enough for you, it's good enough for me.

Reply:
Earlier this year I had populated MM with a set of course keys with the run `3T2019`. However, due to scheduling changes at edx.org, that run didn't make sense any more and the courses on edX were given the run `2T2019a`. I later updated MM with correct course keys, which includes the correct run. 

I'm fine with the idea that the MM courses would be skipped when their keys didn't match the edx.org feed. I just hope that when they are corrected they can be re-ingested in the future.

Reply:
It would be ideal not to have edx_key set on course, you can set course_key on the course run without issue though. If those keys change after being ingested it'll probably require manual cleanup since we're uniquing off the edx ids

Reply:
Just to be explicit in case my earlier explanation wasn't direct, the course and course runs will both be skipped if the edx_key on course is not set. If it is set, the we add the course under the program. If not, the course will be imported only from edx. Although as I mentioned in the previous comment, we use the course/ course run keys as the unique identifiers of those records, so once something is synced, if the keys change it would look like a new course/run to us. I think that means we're OK though since discussions is the only thing using edx_key